### ISSUE TITLE
training not use GPU full perfermance , is just spikes

### ISSUE THREAD

**Issue by amineoui** [opendilab/LightZero#86]
i have everything installed correctly but i not understand why is not use full power of GPU 

![image](https://github.com/opendilab/LightZero/assets/28687360/2922a299-c7f2-491c-b773-7ae08623d11b)

i have some warning but i dont thing is the case ?

![image](https://github.com/opendilab/LightZero/assets/28687360/ca6b1024-c41f-4975-ad46-ffe5bdd9b8ef)

installation is done correctly and nvidia already used for other ai training and it work with full perfermence

![image](https://github.com/opendilab/LightZero/assets/28687360/aaf22481-32f8-48ce-88b8-485bb2393d8f)

please how i can solve this problem , without full use of GPU trainning is take too much time !!

**Comment by amineoui** [comment]
**i remove warnings but issu still there , GPU is used + take some gpu ram but still processor just spikes and** 

warning removed by :
pip3 install numba
use numpy==1.23.0

replace self._env = gym.make('CartPole-v1') by **self._env = gym.make('CartPole-v1', new_step_api=True)** line 32
Comment # self._env.seed(self._seed + np_seed) on cartpole_lightzero_env.py line 45
and # self._env.seed(self._seed) line 49

replace obs = self._env.reset() by **obs = self._env.reset(seed=self._seed)** line 54

replace obs, rew, done, info = self._env.step(action) by **obs, rew, done, _ , info = self._env.step(action)**

this remove all my warning , still have only one related to buffer 
[08-26 15:34:07] WARNING  The data in replay_buffer is not sufficient to sample a mini-batch: batch_size: 256, current buffer statistics is: num_of_all_collected_episodes: 8, num of game segments: 8, number of        train_muzero.py:176
                          transitions: 162 continue to collect now ....
**Please help me to solve this**

**Comment by puyuan1996** [comment]
Hello,

Thank you for your diligent efforts in eliminating warnings within your local environment. In respect to the issue of low GPU utilization and occasional spikes encountered while operating `cartpole_sampled_efficientzero.py`, potential reasons are as follows.

Within LightZero, the MCTS+RL (Monte Carlo Tree Search + Reinforcement Learning) algorithm predominantly encompasses three stages: `collect`, `train`, and `eval`:

- During the `collect` and `eval` stages, the sampled_efficientzero algorithm is utilized where the network inference is conducted mainly at the leaf nodes of the MCTS. This operation is executed on the GPU, while the MCTS search itself is primarily conducted on the CPU. Consequently, throughout the search process, GPU utilization may remain relatively low. Spikes are likely to manifest during the inference at the leaf nodes.

- Throughout the `train` stage, data is sampled from the replay buffer, model inference is executed, the corresponding loss is calculated, and subsequently, backpropagation is performed to update the model. During this stage, GPU utilization is anticipated to be higher. Nevertheless, as the model applied in RL algorithms and the batch size for training are not exceptionally large, the GPU utilization might not reach its maximum capacity of 100%.

In pursuit of enhancing the efficiency of MCTS search and improving GPU utilization, we will delve further into these areas in our forthcoming research. We deeply appreciate your valuable suggestions.

Best Wishes.
